gap/weights.gi

```diff
@@ -0,0 +1,92 @@
+#############################################################################
+##
+##  weights.gi
+##  Copyright (C) 2023                                Raiyan Chowdhury
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+##
+
+#############################################################################
+# 1. Edge Weights
+#############################################################################
+
+InstallGlobalFunction(EdgeWeightedDigraph,
+function(digraph, weights)
+  local digraphVertices, nrVertices, u, outNeighbours,
+  outNeighbourWeights, idx, w;
+
+  if IsDigraph(digraph) then
+      digraph := DigraphCopy(digraph);
+  else
+      digraph := Digraph(digraph);
+  fi;
+
+  # check all elements of weights is a list
+  if not ForAll(weights, IsListOrCollection) then
+      ErrorNoReturn("2nd argument (list) must be a list of lists,");
+  fi;
+
+  digraphVertices := DigraphVertices(digraph);
+
+  nrVertices := Size(digraphVertices);
+
+  # check number there is an edge weight list for vertex u
+  if nrVertices <> Size(weights) then
+      ErrorNoReturn("number of out neighbours and weights must be equal,");
+  fi;
+
+  # check all elements of weights is a list and size/shape is correct
+  for u in digraphVertices do
+      outNeighbours := OutNeighbors(digraph)[u];
+
+      outNeighbourWeights := weights[u];
+
+      # check number of out neigbours for u
+      # and number of weights given is the same
+      if Size(outNeighbours) <> Size(outNeighbourWeights) then
+          ErrorNoReturn(
+              "size of out neighbours and weights for vertex ",
+               u, " must be equal,");
+      fi;
+
+      # check all elements of out neighbours are int
+      for idx in [1 .. Size(outNeighbours)] do
+          w := Float(outNeighbourWeights[idx]);
+
+          if not (IsInt(w) or IsFloat(w) or IsRat(w)) then
```

Comment:
I don't think this checks what you want it to, if `w` is an int prior to line 56, then it is a float in line 58. Same for a rational.  If `w` is a string for example, then presumably line 56 fails. It'd be better to just check `outNeighbourWeights[idx]` directly without the conversion in line 56.